internal/redisclient/redisclient.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2025 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redisclient
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/rond-authz/rond/logging"
+
+	"github.com/redis/go-redis/v9"
+)
+
+type RedisClient struct {
+	client   *redis.Client
+	database int
+}
+
+type ConnectionOpts struct {
+	MaxRetries   int
+	ReadTimeout  time.Duration
+	WriteTimeout time.Duration
+	DialTimeout  time.Duration
+	IdleTimeout  time.Duration
+	PoolSize     int
+	MinIdleConns int
+}
+
+// NewRedisClient tries to setup a new RedisClient instance.
+// The function returns a `nil` client if the environment variable `RedisURL` is not specified.
+func NewRedisClient(logger logging.Logger, redisURL string, connectionOptions ConnectionOpts) (*redis.Client, error) {
+	if redisURL == "" {
+		logger.Info("No Redis configuration provided, skipping setup")
+		return nil, nil
+	}
+
+	logger.Trace("Start Redis client set up")
+
+	// Parse Redis URL
+	parsedURL, err := url.Parse(redisURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed Redis URL validation: %s", err.Error())
+	}
+
+	// Extract database number from URL path
+	database := 0
+	if parsedURL.Path != "" && len(parsedURL.Path) > 1 {
+		dbStr := parsedURL.Path[1:] // Remove leading slash
+		if db, err := strconv.Atoi(dbStr); err == nil {
+			database = db
+		}
+	}
+
+	// Extract username and password from URL
+	username := ""
+	password := ""
+	if parsedURL.User != nil {
+		username = parsedURL.User.Username()
+		password, _ = parsedURL.User.Password()
+	}
+
+	// Build Redis options
+	opts := &redis.Options{
+		Addr:     parsedURL.Host,
+		Username: username,
+		Password: password,
+		DB:       database,
+	}
+
+	// Set connection options if provided
+	if connectionOptions.MaxRetries > 0 {
```

Comment:
these connection options are never used, are you planning on adding more environment variables to map them? otherwise they are dead code and should be removed

Reply:
We can added them later if we need them.